### PR TITLE
Fix permanent redirct issues detected by Repology

### PR DESCRIPTION
**Pull request by mt-caret**
###### Motivation for this change

Trivial search-and-replace for urls that have permanent redirects.
This partially addresses https://github.com/NixOS/nixpkgs/issues/30636.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by peti — DISMISSED** [review]
Please read the very first line of `pkgs/development/haskell-modules/hackage-packages.nix`. That is a *generated file* and any changes you make there manually will be lost during the next update.

**Comment by mt-caret** [comment]
Fixed.

File: pkgs/tools/backup/tarsnapper/remove-argparse.patch [hunk 1]
```diff
@@ -1,7 +1,7 @@
 --- tarsnapper-0.4-src.org/setup.py	1980-01-02 00:00:00.000000000 +0000
 +++ tarsnapper-0.4-src/setup.py	2017-07-16 10:54:36.596499451 +0100
 @@ -45,6 +45,6 @@
-       url='http://github.com/miracle2k/tarsnapper',
```
Comment (globin) [inline pkgs/tools/backup/tarsnapper/remove-argparse.patch]: This will break the patch

File: pkgs/tools/text/unoconv/unoconv-python3.patch [hunk 2]
```diff
@@ -357,10 +357,10 @@ index 30e6706..f72cf08 100755
  #        debug_office()
 -        print >>sys.stderr, "unoconv: Cannot find a suitable office installation on your system."
 -        print >>sys.stderr, "ERROR: Please locate your office installation and send your feedback to:"
--        print >>sys.stderr, "       http://github.com/dagwieers/unoconv/issues"
```
Comment (globin) [inline pkgs/tools/text/unoconv/unoconv-python3.patch]: This will break the patch

**Review by globin — CHANGES_REQUESTED** [review]

**Comment by mt-caret** [comment]
Fixed

**Comment by mt-caret** [comment]
ping @globin @FRidh ?

**Comment by FRidh** [comment]
I think it is ok though I have not verified all changes.

**Comment by mt-caret** [comment]
ping @globin ?

**Comment by FRidh** [comment]
@mt-caret if you rebase, squash the commits, and ping me, then I'll merge it.

**Pull request closed without merging**